src/Entity/Reflection/PropertyMetadata.php

```diff
@@ -140,16 +140,40 @@ public function isValid(& $value)
 
 		foreach ($this->types as $type => $foo) {
 			if ($type === 'datetime') {
-				if ($value instanceof \DateTime || $value instanceof \DateTimeInterface) {
+				if ($value instanceof \DateTime) {
+
 					return TRUE;
-				}
+				} elseif ($value instanceof \DateTimeImmutable) {
+					$value = new \DateTime($value->format('c'));
 
-				if (is_string($value) && $value !== '') {
+					return TRUE;
+				} elseif (is_string($value) && $value !== '') {
 					$value = new \DateTime($value);
 					$value->setTimezone(new DateTimeZone(date_default_timezone_get()));
+
 					return TRUE;
 				} elseif (ctype_digit($value)) {
 					$value = new \DateTime("@{$value}");
+
+					return TRUE;
+				}
+
+			} elseif ($type === 'datetimeimmutable') {
+				if ($value instanceof \DateTimeImmutable) {
+
+					return TRUE;
```

Comment:
please, we write an empty line after return, not before it :) please fix in all these changes :)